frontend/packages/console-dynamic-plugin-sdk/src/extensions/topology-details.ts

```diff
@@ -34,12 +34,8 @@ export type DetailsTabSection = ExtensionDeclaration<
     tab: string;
     /** Returns a section for the graph element or undefined if not provided.
      * SDK component: <Section title={<optional>}>... padded area </Section>
-     * @param renderNull should be used for section that defines Adapter to
-     *  determine if adapter component renders null or not
      * */
-    section: CodeRef<
-      (element: GraphElement, renderNull?: () => null) => React.Component | undefined
-    >;
+    section: CodeRef<(element: GraphElement) => React.Component | undefined>;
```

Comment:
With other extension hooks we define them as `CodeRef<ExtensionHook<React.Component | undefined[]>>` because hooks can be in a `loading` state.
Even if we don't track loading states, the UI will pop in as loading completes. Tabs and sections will appear async. Not sure this is any worse though than having a loading state and waiting.

If we follow the same pattern as other hook extensions, we can still implement the same render logic as we have today but keeps the doors open for future refinement with error and loading handling.